olclient/bots.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+"""
+Classes and methods for bulk edits that match the following pattern:
+1. Search data dump for X condition
+2. Remove or modify record that fit s condition X
+3. Create log of modification
+4. Repeat 1-3 for all records that meet condition X
+"""
+
+
+import argparse
+import datetime
+import logging
+import sys
+
+from olclient.openlibrary import OpenLibrary
+from os import makedirs, path
+
+
+class BaseBot(object):
+    def __init__(self, ol=None, dry_run=True, limit=1) -> None:
+        """Create logger and class variables"""
+        self.ol = ol or OpenLibrary()
+
+        self.parser = argparse.ArgumentParser(description=__doc__)
+        self.parser.add_argument('-f', '--file', type=str, default=None, help='Path to file containing input data')
+        self.parser.add_argument('-l, --limit', type=int, default=1,
+                                 help='Limit number of edits performed on external data.'
+                                      'Set to zero to allow unlimited edits.')
+        self.parser.add_argument('-d', '--dry-run', type=self._str2bool, default=True,
+                                 help='Execute the script without performing edits on external data.')
+        self.args = self.parser.parse_args()
+        self.dry_run = getattr(self.args, 'dry-run', None) or dry_run
+        self.limit = getattr(self.args, 'limit', None) or limit
+        self.changed = 0
+
+        job_name = path.splitext(sys.argv[0])[0]
+        if '/' or '\\' in job_name:  # FIXME: this is disgusting and path.isfile() and path.isdir() don't work
+            job_name = path.splitext(job_name)[-1] or 'a_girl_has_no_name'
```

Comment:
Make this a parameter to the constructor; the subclass should be able to use `__name__` (or whatever it's called) and pass that in without issue.

Reply:
https://docs.python.org/3/library/os.html#os.sep is '/' for POSIX and '\\' for Windows.

Reply:
Done